# Coleccionista

In [25]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

# Importe de Datos
data = pd.read_csv("./Coleccionista.csv")
data.head()

Painting  Price  Bidders  Age
0         1   3470       10   67
1         2   3500        8   56
2         3   3700        7   73
3         4   3860        4   71
4         5   3920       12   99

# 1. Determinar las variables independientes

In [26]:
X = data[['Bidders', 'Age']]
X 

Bidders  Age
0        10   67
1         8   56
2         7   73
3         4   71
4        12   99
5        10   87
6        11   78
7         8   83
8        13   90
9        13   98
10        0   91
11        7   93
12        2   97
13        6   79
14        4   83
15        3   71
16        9   89
17        5  103
18        3  106
19        8   93
20        8   88
21        4   96
22        5   94
23        3   88
24        1   84

# #2. Determinar la variable dependiente

In [27]:
y = data[['Price']]
y

Price
0    3470
1    3500
2    3700
3    3860
4    3920
5    3900
6    3830
7    3940
8    3880
9    3940
10   4200
11   4060
12   4200
13   4020
14   4190
15   4130
16   4130
17   4370
18   4450
19   4390
20   4380
21   4540
22   4660
23   4710
24   4880

In [28]:
# Ajustar el modelo de regresión múltiple
model = LinearRegression()
model.fit(X, y) #Entrenando el modelo
y_pred = model.predict(X) #Predicciones del modelo
y_pred

#Métricas estadísticas
mse = mean_squared_error(y, y_pred) # Error de la media cuadrática
r2 = r2_score(y, y_pred) # Varianza

# 3. Valor de Intersección

In [29]:
model.intercept_

array([3080.05303581])

In [30]:
# Coeficientes de la regresión
print("Coeficientes: ", model.coef_)
print("Error de la media cuadrática (MSE): ", mse)
print("Varianza (R^2): ", r2)

Coeficientes:  [[-54.18854049  16.2891028 ]]
Error de la media cuadrática (MSE):  43396.435062885896
Varianza (R^2):  0.6478019489117818


# 4. Estimación de Precio de Subasta

In [31]:
Y = model.predict(np.array([[14,80]]))
print("Precio estimado de una subasta: ", Y)

Precio estimado de una subasta:  [[3624.54169318]]


c:\Users\panch\anaconda3\Lib\site-packages\sklearn\utils\validation.py:2749: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


# Conclusión 
El valor de 𝑅 a la 2 = 0.648 indica que el modelo explica cerca del 65% de la variación en el precio, por lo que tiene una capacidad explicativa razonable. La antigüedad de la pintura influye positivamente en el precio, mientras que el número de postores presenta una relación inversa. Aunque esto puede parecer sorprendente, es lógico si se considera que las obras más valiosas suelen atraer a pocos compradores especializados, mientras que subastas con muchos postores corresponden a pinturas de menor valor.